### PR TITLE
feat: batcher

### LINKED ISSUE #160: feat: create batcher
Create a new cli tool using rust, that acts as an intermediary between the task sender & the smart contract. It should:
1. take proofs from the task sender using a web socket connection.
2. Verify the proofs
3. Post the data to an S3 bucket
4. Hash the data

### PR DESCRIPTION
**Pull request by taturosati**
# Description

Created a new cli tool, that acts as an intermediary between the task sender & the smart contract. It

1. takes proofs from a test client using a web socket connection.
2. Verifies the proof
3. Hashes the data & returns the hash to the client
4. Posts batches to an S3 bucket

# To test

1. Create a .env file inside the batcher directory (e.g at `batcher/.env.example`). As an alternative you can manually set the aws environment variables as described [here](https://docs.aws.amazon.com/sdkref/latest/guide/environment-variables.html)
2. Run `make batcher_start`
3. Run `make batcher_send_dummy_task` twice
4. See logs to check that data is posted

Closes https://github.com/yetanotherco/aligned_layer/issues/160

### CODE REVIEW

File: batcher/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+
+const S3_BUCKET_NAME: &str = "storage.alignedlayer.com";
+
+// Implement the Listener trait for the App struct
+impl Listener for Arc<App> {
+    fn listen(&self, address: &str) -> impl Future {
+        async move {
+            // Create the event loop and TCP listener we'll accept connections on.
+            let try_socket = TcpListener::bind(address).await;
+            let listener = try_socket.expect("Failed to bind");
```
Comment (entropidelic) [inline batcher/src/lib.rs]: could these lines be condensed like this?
```suggestion
            let listener = TcpListener::bind(address).await.expect("Failed to bind");
```

File: batcher/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+sha3 = "0.10.8"
+aws-config = "1.4.0"
+aws-sdk-s3 = "1.29.0"
+bytes = "1.6.0"
+hex = "0.4.3"
+dotenv = "0.15.0"
+anyhow = "1.0.83"
+
+
+
```
Comment (entropidelic) [inline batcher/Cargo.toml]:
```suggestion
```

File: batcher/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+sp1-sdk = { git = "https://github.com/succinctlabs/sp1.git", rev = "v1.0.0-testnet" }
+bincode = "1.3.3"
+sha3 = "0.10.8"
+aws-config = "1.4.0"
+aws-sdk-s3 = "1.29.0"
+bytes = "1.6.0"
+hex = "0.4.3"
+dotenv = "0.15.0"
+anyhow = "1.0.83"
+
```
Comment (entropidelic) [inline batcher/Cargo.toml]:
```suggestion
```

**Review by entropidelic — APPROVED** [review]

**Review by JuArce — APPROVED** [review]
lgtm

**Pull request merged**